### PR TITLE
WIP: 2.0 docs

### PR DESCRIPTION
**Pull request by Gouthamve**
Super early docs for 2.0.

TODO: 
- [ ] Better explanations for rule-groups
- [ ] Better operating/troubleshooting docs
- [ ] #740 
- [ ] One post about upgrade path
- [ ] Lots more :)

We need to figure out the versioned docs soon now.

@brian-brazil @fabxc @grobie @juliusv @beorn7

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/docs/770)

### CODE REVIEW

**Comment by grobie** [comment]
> We need to figure out the versioned docs soon now.

I'm having a meeting today with nanoc people to see if and what would be available. Will present the results afterwards.

File: content/docs/alerting/rules.md [hunk 1]
```diff
@@ -18,22 +18,22 @@ rules](../../querying/rules).
...
-      [ LABELS <label set> ]
-      [ ANNOTATIONS <label set> ]
+    alert: <alert name>
+    expr: <expression>
+    [ for: <duration> ]
+    [ labels: map[string]string ]
+    [ annotations: map[string]string ]
 
-The optional `FOR` clause causes Prometheus to wait for a certain duration
+The optional `for` clause causes Prometheus to wait for a certain duration
```
Comment (brian-brazil) [inline content/docs/alerting/rules.md]: It's an option now rather than clause.

File: content/docs/introduction/comparison.md [hunk 2]
```diff
@@ -110,8 +110,7 @@ strings, and millisecond resolution timestamps.
 
 InfluxDB uses a variant of a [log-structured merge tree for storage with a
 write ahead log](https://docs.influxdata.com/influxdb/v1.2/concepts/storage_engine/),
-sharded by time. This is much more suitable to event logging than Prometheus's
-append-only file per time series approach.
+sharded by time. This is much more suitable to event logging than Prometheus's approach.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: This is as-suitable now, so I'd say that they're the same.

File: content/docs/introduction/faq.md [hunk 3]
```diff
@@ -223,39 +200,6 @@ a priority right now. Note that a counter, even if incremented
 one million times per second, will only run into precision issues
 after over 285 years.
 
-### Why does Prometheus use a custom storage backend rather than [some other storage method]? Isn't the "one file per time series" approach killing performance?
```
Comment (brian-brazil) [inline content/docs/introduction/faq.md]: We'll still need to keep this in some form, probably explain the history.

File: content/docs/introduction/getting_started.md [hunk 4]
```diff
@@ -224,10 +219,15 @@ Try graphing this expression.
...
 To record the time series resulting from this expression into a new metric
 called `job_service:rpc_durations_seconds_count:avg_rate5m`, create a file
-with the following recording rule and save it as `prometheus.rules`:
-
-```
-job_service:rpc_durations_seconds_count:avg_rate5m = avg(rate(rpc_durations_seconds_count[5m])) by (job, service)
+with the following recording rule and save it as `prometheus.rules.yaml`:
+
+```yaml
+version: 1
```
Comment (brian-brazil) [inline content/docs/introduction/getting_started.md]: The version is gone

File: content/docs/introduction/getting_started.md [hunk 5]
```diff
@@ -224,10 +219,15 @@ Try graphing this expression.
 
 To record the time series resulting from this expression into a new metric
 called `job_service:rpc_durations_seconds_count:avg_rate5m`, create a file
-with the following recording rule and save it as `prometheus.rules`:
-
-```
-job_service:rpc_durations_seconds_count:avg_rate5m = avg(rate(rpc_durations_seconds_count[5m])) by (job, service)
+with the following recording rule and save it as `prometheus.rules.yaml`:
```
Comment (brian-brazil) [inline content/docs/introduction/getting_started.md]: We use .yml elsewhere.

I remain a fan of a `filename.rules`.
This is likely to get confused with `prometheus.yml`, a different example may be better (e.g. `service`).

File: content/docs/introduction/roadmap.md [hunk 6]
```diff
@@ -11,12 +11,6 @@ features and current work, see the issue trackers for the various repositories,
 for example, the [Prometheus
 server](https://github.com/prometheus/prometheus/issues).
 
-### New storage engine
-
-A new more efficient storage engine is in development. It will offer reduced
-resource usage, better inverted indexes, and generally allow Prometheus to
-scale further.
-
 ### Long-term storage
```
Comment (brian-brazil) [inline content/docs/introduction/roadmap.md]: Is is too early to remove this too?

File: content/docs/operating/storage.md [hunk 7]
```diff
@@ -6,55 +6,12 @@ sort_rank: 2
...
-Prometheus has a sophisticated local storage subsystem. For indexes,
-it uses [LevelDB](https://github.com/google/leveldb). For the bulk
-sample data, it has its own custom storage layer, which organizes
-sample data in chunks of constant size (1024 bytes payload). These
-chunks are then stored on disk in one file per time series.
+Prometheus has a sophisticated local storage subsystem. 
 
 This sections deals with the various configuration settings and issues you
-might run into. To dive deeper into the topic, check out the following talks:
+might run into. To dive deeper into the topic, check out the following blogs:
```
Comment (brian-brazil) [inline content/docs/operating/storage.md]: blog.

File: content/docs/operating/storage.md [hunk 8]
```diff
@@ -65,295 +22,17 @@ not what you want for actual operations. The flag `storage.local.retention`
...
-  occupied by the Prometheus process.
-* `go_memstats_alloc_bytes`: Go heap size (allocated objects in use plus allocated
-  objects not in use anymore but not yet garbage-collected).
+* `tsdb_active_appenders`: The current number of active appenders, which shows the ingestion pressure.
+* `tsdb_blocks_loaded`: The number of tsdb blocks loaded.
+* `tsdb_compaction_duration`: A summary for the compaction duration.
+* `tsdb_compactions_failed_total`: The number of failed compactions.
+* `tsdb_compactions_total`: The number of compactions finished.
+* `tsdb_compactions_triggered_total`: The number of compactions triggered.
+* `tsdb_samples_appended_total` The number of samples ingested after startup.
```
Comment (brian-brazil) [inline content/docs/operating/storage.md]: we don't have "after startup" on other counters.

File: content/docs/practices/rules.md [hunk 9]
```diff
@@ -49,70 +49,82 @@ conflicts and give you more useful metrics and alerts.
...
+- name: req-failures
+  rules:
+  - record: instance_path:request_failures:rate5m
+    expr: rate(request_failures_total{job="myjob"}[5m])
+    
+  - record: instance_path:request_failures_per_requests:ratio_rate5m
+    expr: |
     instance_path:request_failures:rate5m{job="myjob"}
-  /
+    /
```
Comment (brian-brazil) [inline content/docs/practices/rules.md]: This is inconsistent with the indentation below, we should keep it all consistent as this file is the primary source for how to format rules.

File: content/docs/querying/rules.md [hunk 10]
```diff
@@ -41,22 +41,38 @@ the original expression every time it is needed. This is especially useful for
...
+```
+  # Saving the per-job HTTP in-progress request count as a new set of time series:
+  - record: job:http_inprogress_requests:sum
+    expr: sum(http_inprogress_requests) by (job)
+   
+  # Drop or rewrite labels in the result time series:
+  - record: new_time_series
+    expr:  old_time_series
+    labels:
+      label_to_change: new_value
```
Comment (brian-brazil) [inline content/docs/querying/rules.md]: I'd call this `label_to_set`, as the label may not be present.

File: content/docs/querying/rules.md [hunk 11]
```diff
@@ -41,22 +41,38 @@ the original expression every time it is needed. This is especially useful for
...
+Some examples:
+```
+  # Saving the per-job HTTP in-progress request count as a new set of time series:
+  - record: job:http_inprogress_requests:sum
+    expr: sum(http_inprogress_requests) by (job)
+   
+  # Drop or rewrite labels in the result time series:
+  - record: new_time_series
+    expr:  old_time_series
+    labels:
```
Comment (brian-brazil) [inline content/docs/querying/rules.md]: It'd be worth making a mention that unlike alerting rules, there's no templating here.

**Comment by fabxc** [comment]
From the nanoc perspective it's probably totally possible. We just need to move all .md files into the version of the repos they are about and have a bash script vendor them int under `<project>/<major>.<minor>/*`.

**Comment by grobie** [comment]
Will need to be done differently.

**Pull request closed without merging**